### PR TITLE
Provide a debug render mode, more intuitive box select, and optimizations in listing transformations

### PR DESCRIPTION
**Pull request by phschaad**
- A boolean flag when initializing the SDFV can force SDFV to draw a coordinate-grid of configurable grid width over the SDFV canvas. Additionally, in that debug mode, mouse coordinates get printed to the bottom right of the canvas, and all elements draw their bounding boxes and center points.

- Box select behaves more intuitively by only selecting elements **INSIDE** the selection box, instead of all elements *intersecting* the bounding box.

- Transformations are not re-sorted every time the mouse moves. They are only re-sorted once the selection and/or viewport changes.

- The sidebar/info-box mentions when multiple elements are selected. This is to avoid ambiguity when selecting multiple elements, as it is otherwise unclear which element's information is being displayed.

### CODE REVIEW

File: renderer.js [hunk 1]
```diff
@@ -1574,6 +1629,7 @@ class SDFGRenderer {
 
     on_mouse_event(event, comp_x_func, comp_y_func, evtype = "other") {
         let dirty = false; // Whether to redraw at the end
+        let resort_transform = false; // Whether to re-sort at the end
```
Comment (tbennun) [inline renderer.js:1632]: The name is too specialized for the vscode plugin. It should be viewport/selection changed or something similar.

File: renderer_elements.js [hunk 2]
```diff
@@ -82,7 +115,7 @@ class SDFG extends SDFGElement {
 }
 
 class State extends SDFGElement {
-    draw(renderer, ctx, mousepos) {
+    draw(renderer, ctx, mousepos, debug_draw = false) {
```
Comment (tbennun) [inline renderer_elements.js:118]: there must be an easier way without overriding every class' draw and draw_simple methods...

**Review by tbennun — CHANGES_REQUESTED** [review]

File: renderer_elements.js (hunk 2 above)
Comment (phschaad) [reply renderer_elements.js:118]: Technically `debug_draw` can be read as a global inside `SDFGElement`'s `debug_draw` method, but I believe that still requires a super call. Alternatively, all the `debug_draw` calls can be explicitly made from within `draw_sdfg`?

Comment (tbennun) [reply renderer_elements.js:118]: The latter may work better, yes.

File: renderer_elements.js [hunk 3]
```diff
@@ -112,18 +147,6 @@ class State extends SDFGElement {
...
-            ctx.beginPath();
-            ctx.moveTo(this.x, this.y - LINEHEIGHT);
-            ctx.lineTo(this.x, this.y + LINEHEIGHT);
-            ctx.stroke();
-            ctx.beginPath();
-            ctx.moveTo(this.x - LINEHEIGHT, this.y);
-            ctx.lineTo(this.x + LINEHEIGHT, this.y);
-            ctx.stroke();
-        }
-
```
Comment (tbennun) [inline renderer_elements.js:126]: why remove this?

**Review by tbennun — CHANGES_REQUESTED** [review]

File: renderer_elements.js (hunk 3 above)
Comment (phschaad) [reply renderer_elements.js]: This is an embarrassing mistake, thanks for catching it! This hunk was reverted.

**Review by tbennun — APPROVED** [review]

**Comment by phschaad** [comment]
Thanks for the review!

**Pull request merged**